### PR TITLE
Test history table for edition and version errors

### PR DESCRIPTION
**Pull request by m-linner-ericsson**
### Applicable Issues

None

### Description of the Change

With the introduction of our event definitions as YAML files, it has enabled us
to generate a manifest file. This has in turn enabled us now to test our
documentation against the manifest. This PR introduces:

- test checking that the edition names in our history tables exist in the
  manifest
- Test checking that the specified edition does contain the versions stated
- It fixes an error found with the tests

### Alternate Designs

 - Don't know if adding the interface for reading the manifest file should reside
in `generate_manifest.py`, but it felt suitable to have them in one place.
 - Running unittests against a real event manifest file has its drawbacks.

### Benefits

We can ensure that our history tables contain valid information.

### Possible Drawbacks

None that I can think of

### Sign-off

Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I have the right
to submit it under the open source license indicated in the file; or

(b) The contribution is based upon previous work that, to the best of my
knowledge, is covered under an appropriate open source license and I have the
right under that license to submit that work with modifications, whether created
in whole or in part by me, under the same open source license (unless I am
permitted to submit under a different license), as indicated in the file; or

(c) The contribution was provided directly to me by some other person who
certified (a), (b) or (c) and I have not modified it.

(d) I understand and agree that this project and the contribution are public and
that a record of the contribution (including all personal information I submit
with it, including my sign-off) is maintained indefinitely and may be
redistributed consistent with this project or the open source license(s)
involved.

Signed-off-by: Mattias Linnér mattias.linner@ericsson.com

### CODE REVIEW

**Review by e-backmark-ericsson — DISMISSED** [review]

**Review by magnusbaeck — COMMENTED** [review]
The underscore-prefixed local variable name looks highly suspect but the other comments are optional.

File: generate_manifest.py [hunk 1]
```diff
@@ -37,6 +38,92 @@
...
+        )
+        self._edition_tag_map = dict()
+        for edition in self._event_manifest:
+            self._edition_tag_map[edition["tag"]] = edition
+
+    def is_edition_tag(self, question_tag: str) -> bool:
+        """
+        Test if the provided tags matches any of the edition tags
+        """
+        return question_tag in self.tags
```
[the remark this answers is not in the document]
Comment (m-linner-ericsson) [reply generate_manifest.py:63]: Maybe, intention from my side was to make it clearer. Do you want me to remove it?

File: generate_manifest.py [hunk 2]
```diff
@@ -37,6 +38,92 @@
...
 
 
+class Manifest:
+    """
+    Python interface as a class for reading the event_manifest.yaml file.
+    """
+
+    def __init__(self, manifest_path: str):
+        with open(manifest_path) as file:
+            _raw_event_manifest = yaml.YAML().load(file)
```
[the remark this answers is not in the document]
Comment (m-linner-ericsson) [reply generate_manifest.py]: Fixed in [bc47d9c](https://github.com/eiffel-community/eiffel/pull/371/commits/bc47d9c9e6b10b862466ea3974ab7b26e3df2e5e)

File: generate_manifest.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (magnusbaeck) [reply generate_manifest.py:64]: No, I don't feel that strongly about it.

**Review by magnusbaeck — APPROVED** [review]

**Review by e-backmark-ericsson — APPROVED** [review]

**Pull request merged**